### PR TITLE
Feature/json module

### PR DESCRIPTION
**Pull request by EluciusFTW**
image

### CODE REVIEW

File: src/spectrecoff-cli/commands/Json.fs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    inherit Command<JsonSettings>()
+    interface ICommandLimiter<JsonSettings>
+
+    override _.Execute(_context, _settings) =
+
+        Many [
+            CO [ 
+                Emoji ":backhand_index_pointing_right:"
+                P "This example is taken directly from Spectre.Console" 
+                Emoji "astonished_face"
```
Comment (DanielMuckelbauer) [inline src/spectrecoff-cli/commands/Json.fs:20]: I don't remember exactly how this `Emoji` union case (right term? :)) came into being, but why not just something like:

```
                E Emoji.Known.BackhandIndexPointingRight
                P "This example is taken directly from Spectre.Console"
                E Emoji.Known.AstonishedFace
```

It seems strange to me to use some more or less magic strings if we have them strongly typed as constants.

File: src/spectrecoff-cli/commands/Panel.fs [hunk 2]
```diff
@@ -33,8 +33,8 @@ type PanelExample() =
         |> panel header 
         |> toConsole    
 
-        P "That surrounding border can be customized easily, e.g., to take up as much horizontal space as needed."
-        |> customPanel { defaultPanelLayout with Sizing = Expand } " Customization " 
+        P "That surrounding border can be customized easily!"
```
Comment (DanielMuckelbauer) [inline src/spectrecoff-cli/commands/Panel.fs:36]: Haha, love how you felt the need to put more emphasis on this :) `!`

**Review by DanielMuckelbauer — CHANGES_REQUESTED** [review]

File: src/spectrecoff-cli/commands/Json.fs (hunk 1 above)
Comment (EluciusFTW) [reply src/spectrecoff-cli/commands/Json.fs:20]: The reason is simple - there is a bug using the constants. Look at this example: If you use the string literal you can use the `:`, or leave them, works both. If you use the constant, the emoji is padded with `:` on both sides :/
image

File: src/spectrecoff/Json.fs [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+let mutable booleanColor = calmColor
+let mutable booleanDecorations = [ Decoration.Bold ]
+
+let mutable nullColor = calmColor
+let mutable nullDecorations = [ Decoration.Dim; Decoration.Strikethrough ]
+
+let private aggregate decorations =
+    decorations |>  List.reduce (|||)
+
+let private applyColors (json: JsonText) =
```
[the remark this answers is not in the document]
Comment (EluciusFTW) [reply src/spectrecoff/Json.fs]: Renamed, thx.

File: src/spectrecoff/Json.fs [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+let mutable stringDecorations = [ Decoration.Bold ]
+
+let mutable numberColor = edgyColor
+let mutable numberDecorations = [ Decoration.Bold ]
+
+let mutable booleanColor = calmColor
+let mutable booleanDecorations = [ Decoration.Bold ]
+
+let mutable nullColor = calmColor
+let mutable nullDecorations = [ Decoration.Dim; Decoration.Strikethrough ]
```
[the remark this answers is not in the document]
Comment (EluciusFTW) [reply src/spectrecoff/Json.fs]: I wanted to have one of the arrays with at least two elements. But I agree, removed it.

File: src/spectrecoff-cli/commands/Json.fs (hunk 1 above)
Comment (DanielMuckelbauer) [reply src/spectrecoff-cli/commands/Json.fs:20]: Aha! That is interesting, and explains why it looked kind of strange when I implemented the grid:

image

Anyway, thanks for explaining

**Review by DanielMuckelbauer — APPROVED** [review]

**Pull request merged**